operator/watchers/cilium_endpoint.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2016-2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package watchers
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	ciliumv2 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/k8s/informer"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+)
+
+const identityIndex = "identity"
+
+var (
+	errNoCE  = errors.New("object has is not a *cilium_v2.CiliumEndpoint meta")
```

Comment:
There is a typo here, you probably want to drop "has"...

Also, I think it'd be nice to use `fmt.Errorf("unexpected object type - found %T, expected %T)", obj, &CiliumEndpoint{})`  for this, and make the error much more meaningful.